src/dash-table/components/ControlledTable/index.tsx

```diff
@@ -831,7 +831,7 @@ export default class ControlledTable extends PureComponent<ControlledTableProps>
         }, column_conditional_tooltips);
 
         return conditionalTooltips.length ?
-            conditionalTooltips[0] :
+            conditionalTooltips.slice(-1)[0] :
```

Comment:
Inverting the conditional tooltip precedence, the last one that fits the criteria wins.